unstoppable-ios-app/domains-manager-ios/Modules/DomainProfile/Public Profile/FollowersView/PublicProfileFollowersViewModel.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  PublicProfileFollowersViewModel.swift
+//  UBTSharing
+//
+//  Created by Oleg Kuplin on 29.08.2023.
+//
+
+import SwiftUI
+
+extension PublicProfileFollowersView {
+    
+    @MainActor
+    final class PublicProfileFollowersViewModel: ObservableObject, ProfileImageLoader, ViewErrorHolder {
+       
+        let domainName: DomainName
+        let socialInfo: DomainProfileSocialInfo
+        
+        @Published var isLoadingPage = false
+        @Published var selectedType: DomainProfileFollowerRelationshipType = .followers {
+            didSet {
+                didChangeRelationshipType()
+            }
+        }
+        @Published var error: Error?
+        private let numberOfFollowersToTake = 40
+        @Published private(set) var followersList: [DomainProfileFollowerDisplayInfo]?
+        private var followersPaginationInfo: FollowersPaginationInfo = .init()
+        @Published private(set)  var followingList: [DomainProfileFollowerDisplayInfo]?
+        private var followingPaginationInfo: FollowersPaginationInfo = .init()
+        
+        var currentFollowersList: [DomainProfileFollowerDisplayInfo]? {
+            getFollowersListFor(type: selectedType)
+        }
+        
+        init(domainName: DomainName, socialInfo: DomainProfileSocialInfo) {
+            self.domainName = domainName
+            self.socialInfo = socialInfo
+        }
+        
+        func loadIconIfNeededFor(follower: DomainProfileFollowerDisplayInfo) {
+            guard follower.icon == nil else { return }
+            let type = selectedType
+            
+            Task {
+                @MainActor
+                func setIcon(_ icon: UIImage?) {
+                    if case .followers = type,
+                       let i = followersList?.firstIndex(where: { $0.domain == follower.domain }) {
```

Comment:
There is a number of repletion of this code
`.firstIndex(where: { $0.domain == follower.domain })`
Since it is low level, why don't we extract it into smth like:
`func getFirstIndexForFollowerDomain()`?

Reply:
Good point 👍